rfq/manager.go

```diff
@@ -357,6 +414,73 @@ func (m *Manager) handleOutgoingMessage(outgoingMsg rfqmsg.OutgoingMsg) error {
 	return nil
 }
 
+// addScidAlias adds a SCID alias to the alias manager.
+func (m *Manager) addScidAlias(aliasMgr lnd.AliasAdder,
+	scidAlias uint64, assetID asset.ID, peer route.Vertex) error {
+
+	// Retrieve all local channels.
+	ctxb := context.Background()
+	localChans, err := m.cfg.ChannelLister.ListChannels(ctxb)
+	if err != nil {
+		return fmt.Errorf("error listing local channels: %w", err)
+	}
+
+	// Filter for channels with the given peer.
+	peerChannels := make([]lndclient.ChannelInfo, 0)
+	for _, localChan := range localChans {
+		if localChan.PubKeyBytes == peer {
+			peerChannels = append(peerChannels, localChan)
+		}
+	}
+
+	// Identify the correct channel to use as the base SCID for the alias
+	// by inspecting the asset data in the custom channel data.
+	var (
+		assetIDStr = assetID.String()
+		baseSCID   uint64
+	)
+
+	for _, localChan := range peerChannels {
+		var assetData JsonAssetChannel
+		err = json.Unmarshal(localChan.CustomChannelData, &assetData)
+		if err != nil {
+			log.Warnf("Unable to unmarshal channel asset data: %v",
+				err)
+			continue
+		}
+
+		for _, channelAsset := range assetData.Assets {
+			gen := channelAsset.AssetInfo.AssetGenesis
+			if gen.AssetID == assetIDStr &&
+				localChan.PubKeyBytes == peer {
+
+				baseSCID = localChan.ChannelID
+				break
+			}
+		}
+	}
+
+	// As a fallback, if the base SCID is not found and there's only one
+	// channel with the target peer, assume that the base SCID corresponds
+	// to that channel.
+	if baseSCID == 0 && len(peerChannels) == 1 {
```

Comment:
When there are multiple channels available and the fallback mechanism is used, we should fail instead of selecting the first channel. This avoids flaky tests that depend on the arbitrary order of channels.

Reply:
> not sure why we would ever run in this edge case, but why not just fail at this point?

@GeorgeTsagk We can't currently setup channels with custom channel data in the tap itests. This fallback is a workaround for that limitation.